internal/lint/subscribe.go

```diff
@@ -0,0 +1,128 @@
+package lint
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/bsm/bps"
+	"github.com/onsi/ginkgo"
+	Ω "github.com/onsi/gomega"
+)
+
+// SubscriberInput for the shared test.
+type SubscriberInput struct {
+	// Subject holds a subscriber to be tested.
+	Subject bps.Subscriber
+	// SeededMessages contain all the messages, seeded for this subscriber.
+	// Must contain at 2+ messages.
+	// Seed/consume order must be preserved.
+	SeededMessages []bps.SubMessage
+	// Setup is called before testing the subscriber.
+	Setup func(topic string) error
+	// Teardown is called after testing the subscriber.
+	Teardown func(topic string) error
+}
```

Comment:
After I've tried to use this for in-mem subscriber test, I think it would be better to have it like:

```go
type SubscriberInput struct {
  Subject func(topic string, messages []bps.SubMessage) bps.Subscriber // no need for error here, let caller to protect with Expect
}
```

So caller will set up / instantiate subscriber with seeded messages so no need to have SeededMessages/Setup/Teardown (just delegate it to caller).

Reply:
Doing it ^^